src/java/org/apache/cassandra/repair/autorepair/AutoRepairConfig.java

```diff
@@ -0,0 +1,390 @@
+        // The percentage of nodes in the cluster that run repair parallelly. If parallelrepaircount is set, it will choose the larger value of the two.
+        //The problem with a fixed number of nodes (the above property) is that in a large-scale environment,
+        // the nodes keep getting added/removed due to elasticity, so if we have a fixed number, then manual interventions would increase because, on a continuous basis,operators would have to adjust to meet the SLA.
+        //The default is 3%, which means that 3% of the nodes in the Cassandra cluster would be repaired in parallel.
+        // So now, if a fleet, an operator won't have to worry about changing the repair frequency, etc., as overall repair time will continue to remain the same even if nodes are added or removed due to elasticity.
+        // Extremely fewer manual interventions as it will rarely violate the repair SLA for customers
+        public volatile Integer parallel_repair_percentage_in_group;
+        // the upper threshold of SSTables allowed to participate in a single repair session
+        // Threshold to skip a table if it has too many sstables. The default is 10000. This means, if a table on a node has 10000 or more SSTables, then that table will be skipped.
+        // This is to avoid penalizing good neighbors with an outlier.
+        public volatile Integer sstable_upper_threshold;
+        // the minimum time in hours between repairs of the same token range
+        // The minimum number of hours to run one repair cycle is 24 hours. The default is 24 hours.
+        // This means that if auto repair finishes one round on one cluster within 24 hours, it won’t start a new round.
+        // This is applicable for extremely tiny clusters, say 3 nodes.
+        public volatile Integer min_repair_interval_in_hours;
+        // specifies a denylist of datacenters to repair
+        // This is useful if you want to completely avoid running repairs in one or more data centers. By default, it is empty, i.e., the framework will repair nodes in all the datacenters.
+        public volatile Set<String> ignore_dcs;
+        // Set this 'true' if AutoRepair should repair only the primary ranges owned by this node; else, 'false'
+        // It is the same as -pr in nodetool repair options.
+        public volatile Boolean repair_primary_token_range_only;
+        // By default, the value is empty, which means all nodes are in one ring and the repair in that ring should be done node by node.
+        // Adding this config for special cases where certain keyspace data is only stored in
+        // certain data centers. We can run multiple node repairs at the same time.
+        // For example, if the value is “dc1,dc2|dc3|dc4|dc5,dc6,dc7”, there will be 4 groups {dc1, dc2}, {dc3}, {dc4} and {dc5, dc6, dc7}.
+        // This means we can run repair parallely on 4 nodes, each in one group.
+        public volatile Set<String> repair_dc_groups;
```

Comment:
Since this is a global setting, I am not sure this is safe or necessary. Other keyspaces not in an island configuration to a single DC will not get repaired correctly (e.g., system_distributed, system_auth). For single keyspaces that are not replicated, there are already checks on a node if it is not replicated for a keyspace in checkNodeContainsKeyspaceReplica of AutoRepairUtils. Could simplify things a little bit and get rid of the pid in schema?

Reply:
Yeah, it is not necessary. I have removed it.